electrumsv/tests/test_wallet.py

```diff
@@ -486,128 +508,346 @@ def test_multisig(self, mock_app_state, tmp_storage) -> None:
         check_create_keys(wallet, account)
 
 
-@pytest.mark.parametrize("storage_info",
-    get_categorised_files2(TEST_WALLET_PATH, exclude_suffix=".json"))
-@unittest.mock.patch('electrumsv.wallet.app_state', new_callable=_create_mock_app_state)
-def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorageInfo,
-        caplog) -> None:
-    password = initial_password = "123456"
-    password_token = PasswordToken(password)
-    mock_wallet_app_state.credentials.get_wallet_password = lambda wallet_path: password
-
-    # When a wallet is composed of multiple files, we need to know which to load.
-    wallet_filenames = []
-    if storage_info.kind != StorageKind.DATABASE:
-        wallet_filenames.append(storage_info.filename)
-    if storage_info.kind in (StorageKind.DATABASE, StorageKind.HYBRID):
-        wallet_filenames.append(storage_info.filename + DATABASE_EXT)
+def load_wallet(storage: WalletStorage) -> Wallet | None:
+    try:
+        return Wallet(storage)
+    except FileNotFoundError as e:
+        if sys.version_info[:3] >= (3, 8, 0):
+            msg = "Could not find module 'libusb-1.0.dll' (or one of its dependencies)."
+            if msg in e.args[0]:
+                pytest.xfail("libusb DLL could not be found")
+                return
+        raise e
+    except OSError as e:
+        if sys.version_info[:3] < (3, 8, 0):
+            if "The specified module could not be found" in e.args[1]:
+                pytest.xfail("libusb DLL could not be found")
+                return
+        raise e
+
+
+class ReEncryptionTestState(TypedDict):
```

Comment:
1. This state should not be based on a TypedDict. It should be a dataclass. We're not writing javascript.
2. This state has nothing to do with reencryption. It is for tracking wallet test coverage for all possible things we want to track are tested.